lib/utils/source_ip_check.ts

```diff
@@ -116,25 +118,36 @@ export function checkSourceIpForHandler<P1, P2, P3, P4, P5, P6, O>(
   p4: P4,
   p5: P5,
   p6: P6
-) => Promise<O | IResponseErrorForbiddenNotAuthorized> {
+) => Promise<
+  O | IResponseErrorForbiddenNotAuthorized | IResponseErrorInternal
+> {
   return (p1: P1, p2: P2, p3: P3, p4: P4, p5: P5, p6: P6) => {
-    // extract the x-forwarded-for header and the allowed cidrs from the params
-    const x = extractor(p1, p2, p3, p4, p5, p6);
-    const maybeClientIp = x.e1;
-    const cidrs = x.e2;
+    return new Promise(resolve => {
+      // extract the x-forwarded-for header and the allowed cidrs from the params
+      const x = extractor(p1, p2, p3, p4, p5, p6);
+      const maybeClientIp = x.e1;
+      const cidrs = x.e2;
 
-    if (
-      // we have a client IP and...
-      isSome(maybeClientIp) &&
-      // ...either no CIDRs setting or client IP is contained in allowed CIDRs
-      (cidrs.size === 0 || isContainedInCidrs(maybeClientIp.value, cidrs))
-    ) {
-      // forward request to handler
-      return handler(p1, p2, p3, p4, p5, p6);
-    } else {
-      // respond with Not Authorized
-      return Promise.resolve(ResponseErrorForbiddenNotAuthorized);
-    }
+      if (isNone(maybeClientIp)) {
+        return resolve(
+          ResponseErrorInternal(
+            "IP address cannot be extracted from the request"
+          )
+        );
+      }
+
+      if (
```

Comment:
I've kept the actual behavior (fail when no IP is found), tell me if you want to change it 
(pass the check if authorized_cidrs == 0 even when IP is null)